pkg/pgmodel/querier/query_builder.go

```diff
@@ -52,39 +62,52 @@ func BuildSubQueries(matchers []*labels.Matcher) (*clauseBuilder, error) {
 
 		switch m.Type {
 		case labels.MatchEqual:
-			if m.Name == pgmodel.MetricNameLabelName {
+			switch m.Name {
+			case pgmodel.MetricNameLabelName:
 				cb.SetMetricName(m.Value)
-				continue
-			}
-			if m.Name == pgmodel.SchemaNameLabelName {
+			case pgmodel.SchemaNameLabelName:
 				cb.SetSchemaName(m.Value)
-				continue
-			}
-			if m.Name == pgmodel.ColumnNameLabelName {
+			case pgmodel.ColumnNameLabelName:
 				cb.SetColumnName(m.Value)
-				continue
-			}
-			sq := subQueryEQ
-			if matchesEmpty {
-				sq = subQueryEQMatchEmpty
+			default:
+				sq := subQueryEQ
+				if matchesEmpty {
+					sq = subQueryEQMatchEmpty
+				}
+				err = cb.addClause(sq, m.Name, m.Value)
 			}
-			err = cb.addClause(sq, m.Name, m.Value)
 		case labels.MatchNotEqual:
 			sq := subQueryNEQ
 			if matchesEmpty {
 				sq = subQueryNEQMatchEmpty
 			}
 			err = cb.addClause(sq, m.Name, m.Value)
 		case labels.MatchRegexp:
+			re2 := re2Regex.MatchString(m.Value)
```

Comment:
Why do we try to determine if the regex should be treated as an RE2 regex? Why not always use the `_prom_ext.re2_match` function?

Reply:
We could and it would simplify the code... I guess this is trying to minimize the exposure to the new regex matcher until proven its stable enough.

Maybe the best way forward is to leave as is and add a TODO to revisit this once this is out.

Reply:
Added a TODO for the future to reduce complexity.

Reply:
How do you usually handle this kind of things? Would it make sense to release with the current engine as default and have the new one under a feature flag or config option? At a latter release we could make the new one the default. 

That way there's a clearer action for customers if there's an issue with the new engine that we haven't spotted.

Reply:
No, we want to fix this for all the users ASAP and it makes no sense to run with the current engine since it doesn't cover all the necessary use cases. 

Keep in mind, the new engine handles a minority of the cases and vast majority will end up just using the current engine without noticing anything different.